scripts/irods/test/test_federation.py

```diff
@@ -1584,3 +1601,97 @@ def test_peps_for_parallel_mode_transfers__issue_5017(self):
+    # in otherZone. The following users will appear:
+    #
+    #   - admin#otherZone
+    #   - zonehopper#otherZone
+    #   - zonehopper#tempZone (created by setUp())
+
+    plugin_name = IrodsConfig().default_rule_engine_plugin
+
+    def setUp(self):
+        super(Test_Delay_Rule_Removal, self).setUp()
+
+        # session.make_sessions_mixin() creates admins and users that connect to the host
+        # identified by lib.get_hostname().
+        #
+        # For this particular set of tests, that means these users connect to the host where
+        # "otherZone" runs.
+        self.local_admin = self.admin_sessions[0] # admin#otherZone
+        self.local_user = self.user_sessions[0]   # zonehopper#otherZone
+
+        # Create session for zonehopper#tempZone. The session will be connected to tempZone.
+        password = test.settings.FEDERATION.RODSUSER_NAME_PASSWORD_LIST[0][1]
+        host = test.settings.FEDERATION.REMOTE_HOST
+        zone = test.settings.FEDERATION.REMOTE_ZONE
+        self.remote_user = session.make_session_for_existing_user(self.local_user.username, password, host, zone)
+
+        # Create zonehopper#tempZone in otherZone. This must be handled by the test suite rather
+        # than the session.make_sessions_mixin() because that function only knows about the local zone.
+        self.remote_user_home_collection = self.remote_user.remote_home_collection(self.local_user.zone_name)
+        self.local_admin.assert_icommand(['iadmin', 'mkuser', self.remote_user.qualified_username, 'rodsuser'])
+
+        # Validity check: If the remote user's home collection does not exist in the zone these tests
+        # are run from, then the testing environment is in a bad state.
+        self.local_admin.assert_icommand(['ils', self.remote_user_home_collection], 'STDOUT', [self.remote_user.qualified_username])
+
+        # Make sure there are no left over delay rules.
+        self.local_admin.run_icommand(['iqdel', '-a'])
+
+    def tearDown(self):
+        self.remote_user.__exit__()
+        self.local_admin.run_icommand(['iadmin', 'rmuser', self.remote_user.qualified_username])
+        super(Test_Delay_Rule_Removal, self).tearDown()
+
+    @unittest.skipIf(plugin_name == 'irods_rule_engine_plugin-python', 'Skip if testing the PREP')
+    def test_local_zone_user_is_not_allowed_to_delete_delay_rules_created_by_remote_user_with_the_same_username__issue_6482(self):
+        try:
+            with temporary_core_file() as core_re:
+                # Users in a federated environment are not allowed to create delay rules in the
+                # remote zone directly. Remote users can only create delay rules indirectly
+                # (e.g. through a PEP in the federated zone).
+                core_re.add_rule(dedent('''
+                    pep_api_touch_pre(*a, *b, *c)
+                    {
+                        delay("<INST_NAME>irods_rule_engine_plugin-irods_rule_language-instance</INST_NAME>") {
```

Comment:
Should we put a `PLUSET` of some very large number so that the rule doesn't execute and leave the queue? There is a highly unlikely but theoretically possible (I think?) scenario in which the remote delay server executes the delay rule before we can fetch it from the queue on 1686 which could lead to a false negative.

Reply:
yes, put it farther into the future please.

Reply:
Done